.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,28 @@
+name: Golang lint, vet and unit test pipeline
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: github (govet, golint and gotest)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.0 # don't bump this until all operators use 1.18
```

Comment:
what about director-operator v1.2.x which is still on go 1.16.x. anything we need to consider for that?

Reply:
@stuggi good call. pushed a 2nd set of test that uses v1.2.0 so we don't break it on the older golang.